FreeRTOS/FreeRTOS-Plus-TCP
# Fix for the incoming IPv4 destination IP check

Description
-----------
When `ipconfigETHERNET_DRIVER_FILTERS_PACKETS` is disabled, the stack needs to filter out packets based on their destination IP. In reality it doesn't.

Test Steps
-----------
1. ```#define ipconfigETHERNET_DRIVER_FILTERS_PACKETS (0)```
2. ```#define ipconfigETHERNET_DRIVER_FILTERS_FRAME_TYPES (1)``` Note below.
3. Modify your network driver to allow all packets ( promiscuous mode ).
4. Inject an IPv4 packet with any unicast source and unicast destination addresses. Such packets can an will occur in the wild even with modern switches.
5. Note that `prvAllowIPPacketIPv4()` returns `eProcessBuffer` for the injected packet and processing continues as if the packet was destined for the DUT.
6. Once the change from this PR is applied, the DUT correctly drops the packets for which it does not have an end-point.

Random Thoughts
----------
I don't know where that NULL check came from but as of right now, any received packet will have a non-NULL `pxNetworkBuffer->pxEndPoint` as it propagates through the stack.

Note on: I understand that in promiscuous mode, `ipconfigETHERNET_DRIVER_FILTERS_FRAME_TYPES` should probably be disabled, however the default is ipconfigENABLED and the IP layer should be capable of filtering out packets regardless of whether the lower layers did their job or not.

Let me know what you think


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
